backend/sqs_messages.py

```diff
@@ -0,0 +1,66 @@
+import json
+from logging import getLogger
+
+import boto3
+from django.conf import settings
+
+from domain_events.handler import HANDLERS
+
+logger = getLogger(__name__)
+
+
+def process_sqs_messages(event):
+    # Very basic SQS handling
+    # Nothing is loaded so you can't use django in the handlers
+    for record in event["Records"]:
+        if record["eventSource"] != "aws:sqs":
+            logger.info(
+                "Skipping message_id=%s because it is not coming from SQS",
+                record["messageId"],
+            )
+            continue
+
+        process_message(record)
+
+
+def process_message(record):
+    message_id = record["messageId"]
+    receipt_handle = record["receiptHandle"]
+
+    message_attributes = record["messageAttributes"]
+    message_type = message_attributes["MessageType"]["stringValue"]
+
+    handler = HANDLERS.get(message_type, None)
+
+    if not handler:
+        logger.info(
+            "Received SQS message_id=%s message_type=%s but no handler accepts it",
+            message_id,
+            message_type,
+        )
+        return
+
+    try:
+        data = json.loads(record["body"])
+        handler(data)
+    except Exception as exc:
+        # In future we should re-schedule the message with a delay if it fails
```

Comment:
At work I saw we use `tenacity.Retrying` https://tenacity.readthedocs.io/en/latest/

Reply:
do they support SQS? might be worth having a look in the future

Reply:
Well, this will retry executing the handler whenever an Exception occurs, it will not reschedule the task in SQS.
In the docs it's used only with the decorator, I saw used as a context manager as well, think it's the same...

Reply:
uhm ok, my idea was to return them to the queue and process them after a few hours (e.g if slack is down) but it highly depends on what the task is doing so I can see why it retries in-place